### PR TITLE
add queuelistener for logging

### PR DESCRIPTION
**Pull request by ekneg54**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 ### Improvements
 
 * remove logger from Components and Factory signatures
+* add `LogprepMPQueueListener` to outsource logging to a separate process
+* add a single `Queuehandler` handler to root logger to ensure all logs were handled by `LogprepMPQueueListener`
```
Comment (dtrai2) [inline CHANGELOG.md:14]:
```suggestion
* add a single `Queuehandler` to root logger to ensure all logs were handled by `LogprepMPQueueListener`
```

File: logprep/runner.py [hunk 2]
```diff
@@ -151,6 +151,8 @@ def start(self):
         self._logger.info("Shutting down")
         self._manager.stop()
         self._logger.info("Shutdown complete")
+        if self._manager.loghandler is not None:
+            self._manager.loghandler.stop()
```
Comment (dtrai2) [inline logprep/runner.py:155]:
```suggestion
        if self._manager.loghandler is not None:
            self._manager.loghandler.stop()
        self._logger.info("Shutdown complete")
```
I hope we can still use the regular logger after the loghandler was stopped.

File: tests/unit/util/test_logging.py [hunk 3]
```diff
@@ -8,12 +12,42 @@
...
+        logger = logging.getLogger("console")
+        assert isinstance(logger.handlers[0].formatter, LogprepFormatter)
 
-    def test_default_log_config(self):
-        logging.config.dictConfig(DEFAULT_LOG_CONFIG)
-        logger = logging.getLogger("test")
-        assert isinstance(logger.root.handlers[0].formatter, LogprepFormatter)
+    def test_root_logger_has_only_quehandler(self):
+        logger = logging.getLogger("root")
+        assert isinstance(logger.handlers[0], logging.handlers.QueueHandler)
```
Comment (dtrai2) [inline tests/unit/util/test_logging.py:28]: according to the test title you also need one more assertions
```suggestion
        assert len(logger.handlers) == 1, "only one handler should be set"
        assert isinstance(logger.handlers[0], logging.handlers.QueueHandler)
```

**Review by dtrai2 — CHANGES_REQUESTED** [review]

File: logprep/runner.py (hunk 2 above)
Comment (ekneg54) [reply logprep/runner.py:155]: no ... that's the reason, this won't work. After stopping the handler, no logs will be handled bei the console handler

**Review by dtrai2 — APPROVED** [review]

**Pull request merged**